src/pool/srv_target.c

```diff
@@ -1263,9 +1261,13 @@ ds_pool_stop(uuid_t uuid)
 
 	ds_rebuild_abort(pool->sp_uuid, -1, -1, -1);
 	ds_migrate_stop(pool, -1, -1);
+
+	pool_child_delete_all(pool);
```

Comment:
If the sponsor of ds_pool_stop() is not the last user for the ds_pool, then the other reference holders may still use related pool_child. Under such case, is it possible to cause some race?

Reply:
Yes, I thought about this question too. I think if there's no other bug, there should be no other user of this `ds_pool` object. Moreover, if there were such a user, there was no `ds_pool_child` pointer in `ds_pool`. So, although that user might encounter some error, he should not encounter serious problems like segfaults.

Reply:
The race may happen, for example, during `ds_pool_tgt_map_update`(), someone triggers `ds_pool_stop`() as to some `pool_child` is released before collective tasks for `ds_pool_tgt_map_update` completed, such as `dtx_resync` may get failure. It is risky for that because `dtx_resync` is global operation, if miss some `pool_child`, it will affect DTX check result as to the decision. Further more, if the DTX leader wants to commit or abort some DTX, but related pool shard is closed, then it may leave orphan DTX entry on related pool shard. If the pool will be destroyed, it's OK; otherwise, it will be trouble. I am not aware of some know segmentation under such case, but really not sure for that.

Reply:
I think I can pull in one more change originally planned for future work: Wait for all other `ds_pool` references to be released. If we do that before calling `pool_child_delete_all`, the situation should become easier for other `ds_pool` reference holders.

The `dtx_resync` concern sounds like a different problem. Let's chat to understand where the problem is.

Reply:
The root reason for `dtx_resync` trouble is not related with the patch, but the changes in this patch will increase the risk to expose the `dtx_resync` trouble.